src/credentials/KeyManager.h

```diff
@@ -0,0 +1,201 @@
+static constexpr size_t kIteratorsMax = CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS;
+
+// typedef struct GroupOperationalCredentials
+// {
+//     /// Validity start time in microseconds since 2000-01-01T00:00:00 UTC ("the Epoch")
+//     uint64_t start_time;
+//     /// Session Id
+//     uint16_t hash;
+//     /// Operational group key
+//     uint8_t encryption_key[Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
+//     /// Privacy key
+//     uint8_t privacy_key[Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
+// } GroupOperationalCredentials;
+
+// An EpochKey is a single key usable to determine an operational group key
+struct EpochKey
+{
+    static constexpr size_t kLengthBytes = Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES;
+    // Validity start time in microseconds since 2000-01-01T00:00:00 UTC ("the Epoch")
+    uint64_t start_time;
+    // Actual key bits. Depending on context, it may be a raw epoch key (as seen within `SetKeySet` calls)
+    // or it may be the derived operational group key (as seen in any other usage).
+    uint8_t key[kLengthBytes];
+
+    void Clear()
+    {
+        start_time = 0;
+        Crypto::ClearSecretData(&key[0], sizeof(key));
+    }
+};
+
+// A operational group key set, usable by many GroupState mappings
+struct KeySet
+{
+    static constexpr size_t kEpochKeysMax = 3;
+
+    KeySet() = default;
+    KeySet(uint16_t id, SecurityPolicy policy_id, uint8_t num_keys) : keyset_id(id), policy(policy_id), num_keys_used(num_keys)
+    {}
+
+    // Logical id provided by the Administrator that configured the entry
+    uint16_t keyset_id = 0;
+    // Security policy to use for groups that use this keyset
+    SecurityPolicy policy = SecurityPolicy::kCacheAndSync;
+    // Number of keys present
+    uint8_t num_keys_used = 0;
+    // The actual keys for the group key set
+    EpochKey epoch_keys[kEpochKeysMax];
+
+    bool operator==(const KeySet & other) const
+    {
+        VerifyOrReturnError(this->policy == other.policy && this->num_keys_used == other.num_keys_used, false);
+        return !memcmp(this->epoch_keys, other.epoch_keys, this->num_keys_used * sizeof(EpochKey));
+    }
+
+    void ClearKeys()
+    {
+        for (size_t key_idx = 0; key_idx < kEpochKeysMax; ++key_idx)
+        {
+            epoch_keys[key_idx].Clear();
+        }
+    }
+};
+
+struct GroupSession
+{
+    GroupSession()   = default;
+    FabricIndex fabric_index;
+    GroupId group_id = kUndefinedGroupId;
+    SecurityPolicy security_policy;
+    Crypto::SymmetricKeyContext * keyContext = nullptr;
+};
+
+// using KeyContext           = chip::Crypto::SymmetricKeyContext;
+using KeySetIterator       = CommonIterator<KeySet>;
+using KeyContextIterator   = CommonIterator<chip::Crypto::SymmetricKeyContext*&>;
+using GroupSessionIterator = CommonIterator<GroupSession>;
+static constexpr KeysetId kIdentityProtectionKeySetId = 0;
+
+class KeyManager
```

Comment:
chip::Credentials::KeyManager is a very generic name for what this does.

It's managing symmetric keys, not all keys.  And even there, it's managing them in a particular way (implementing the spec's concept of keysets).

This should be named something more specific.